sdk/core/core/src/az_client_secret_credential.c

```diff
@@ -135,13 +147,50 @@ static AZ_NODISCARD az_result az_token_credential_get_token(
         auth_url));
 
     AZ_RETURN_IF_FAILED(az_http_request_builder_add_body(&hrb, auth_body));
-    AZ_RETURN_IF_FAILED(az_http_client_send_request_and_get_body(&hrb, &response_buf));
+
+    static az_http_pipeline pipeline = {
+    .policies = {
+      { .pfnc_process = no_op_policy, .data = NULL },
+      { .pfnc_process = no_op_policy, .data = NULL },
+      { .pfnc_process = no_op_policy, .data = NULL },
+      { .pfnc_process = no_op_policy, .data = NULL },
+      { .pfnc_process = no_op_policy, .data = NULL },
+      { .pfnc_process = no_op_policy, .data = NULL },
+      { .pfnc_process = az_http_pipeline_policy_transport, .data = NULL },
+      { .pfnc_process = NULL, .data = NULL },
+    },
+    };
+
+    AZ_RETURN_IF_FAILED(az_http_pipeline_process(&hrb, &response_buf, &pipeline));
+  }
+
+  az_span body = { 0 };
+  {
+    az_span const response = az_mut_span_to_span(response_buf);
+    az_http_response_parser parser = { 0 };
+    AZ_RETURN_IF_FAILED(az_http_response_parser_init(&parser, response));
+
+    az_http_response_status_line status_line = { 0 };
+    AZ_RETURN_IF_FAILED(az_http_response_parser_read_status_line(&parser, &status_line));
+    if (status_line.status_code != AZ_HTTP_STATUS_CODE_OK) {
+      return AZ_ERROR_HTTP_PAL;
+    }
+
+    while (true) {
+      az_pair header = { 0 };
+      az_result const result = az_http_response_parser_read_header(&parser, &header);
+      if (result == AZ_ERROR_ITEM_NOT_FOUND) {
+        break;
+      }
+      AZ_RETURN_IF_FAILED(result);
+    }
```

Comment:
I believe it would be good to add a function `az_http_response_parser_offset_headers()` to `az_http_reponse_parser.h` to move reader up to body without the az_pair logic.  
Then we can have another function to directly read body that first calls this offset and then the read_body.

I think getting header by header is a little waste.

Reply:
Agree, I'll follow up with Sergey om the best approach.

Reply:
Done